src/openrct2-ui/windows/Network.cpp

```diff
@@ -0,0 +1,409 @@
+{
+    constexpr const char* SizeTable[] = { "B", "KB", "MB", "GB", "TB" };
+
+    double size = bytes;
+    size_t idx = 0;
+    while (size >= 1024.0)
+    {
+        size /= 1024.0;
+        idx++;
+    }
+
+    sprintf(buf, "%.03f %s", size, SizeTable[idx]);
+}
+
+static void window_network_information_invalidate(rct_window* w)
+{
+    window_network_set_pressed_tab(w);
+    window_network_anchor_border_widgets(w);
+    window_align_tabs(w, WIDX_TAB1, WIDX_TAB1);
+}
+
+static void window_network_draw_graph(
+    rct_window* w, rct_drawpixelinfo* dpi, int32_t x, int32_t y, int32_t height, int32_t width, bool received)
+{
+    uint32_t dataMax = 0;
+    for (size_t i = 0; i < _networkHistory.size(); i++)
+    {
+        const NetworkHistory_t& history = _networkHistory[i];
+        for (int n = 0; n < NETWORK_STATISTICS_GROUP_MAX; n++)
+        {
+            if (received)
+                dataMax = std::max<uint32_t>(history.deltaBytesReceived[n], dataMax);
+            else
+                dataMax = std::max<uint32_t>(history.deltaBytesSent[n], dataMax);
+        }
+    }
+
+    uint32_t curY = y;
+
+    // Draw graph.
+    for (size_t i = 0; i < _networkHistory.size(); i++)
+    {
+        NetworkHistory_t history = _networkHistory[i];
+        // std::sort(history.deltaBytesReceived.begin(), history.deltaBytesReceived.end(), std::greater<uint16_t>());
+
+        // NOTE: Capacity is not a mistake, we always want the full length.
+        uint32_t curX = ((float)i / (float)_networkHistory.capacity()) * width;
+
+        for (int n = 0; n < NETWORK_STATISTICS_GROUP_MAX; n++)
+        {
+            uint32_t lineHeight;
+            if (received)
+                lineHeight = ((float)history.deltaBytesReceived[n] / (float)dataMax) * height;
+            else
+                lineHeight = ((float)history.deltaBytesSent[n] / (float)dataMax) * height;
+
+            gfx_draw_line(dpi, x + curX, curY + height, x + curX, curY + height - lineHeight, NetworkTrafficGroupColors[n]);
+        }
+    }
+
+    // Draw box.
+    gfx_draw_line(dpi, x, y, x, y + height, COLOUR_BLACK);
+    gfx_draw_line(dpi, x, y + height, x + width, y + height, COLOUR_BLACK);
+
+    gfx_draw_line(dpi, x, y, x + width, y, COLOUR_BLACK);
+    gfx_draw_line(dpi, x + width, y, x + width, y + height, COLOUR_BLACK);
+}
+
+static void window_network_information_paint(rct_window* w, rct_drawpixelinfo* dpi)
+{
+    window_draw_widgets(w, dpi);
+    window_network_draw_tab_images(w, dpi);
+
+    constexpr int32_t borderGap = 8;
+
+    rct_drawpixelinfo clippedDPI;
+    if (clip_drawpixelinfo(&clippedDPI, dpi, w->x, w->y, w->width, w->height))
+    {
+        dpi = &clippedDPI;
+
+        int32_t x = borderGap;
+        int32_t y = 50;
+
+        char textBuffer[128];
+
+        {
+            sprintf(
+                textBuffer, "%u %*s %.2f B/s", _networkCurrentStats.deltaBytesReceived[NETWORK_STATISTICS_GROUP_TOTAL], 4, "",
```

Comment:
I'm afraid "b/s" should be translatable as well... The French use octets rather than bytes. :-)

Reply:
Learned something new, and seriously?

Reply:
In general, you should use OpenRCT2-strings with placeholders rather than sprintf.

Reply:
I didn't think that it would matter this much in such case but turns out I'm wrong.

Reply:
We should probably have separate strings for kB, MB etc… Then standardise their use through the code base, e.g. in string 5759 (`Downloading map ... ({INT32} / {INT32}) KiB`).